src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+    Collections.sort(listified, new Comparator<Map.Entry<String, ReportItem>>() {
+      @Override
+      public int compare(Map.Entry<String, ReportItem> s1, Map.Entry<String, ReportItem> s2) {
+        return s1.getValue().getStock().getName().compareTo(s2.getValue().getStock().getName());
+      }
+    });
+    return listified;
+  }
+
+  private static boolean hasDnaAnalyteName(SampleDto dto) {
+    return (dto.getName().endsWith("_D") || dto.getName().contains("_D_"))
+        && !dto.getSampleType().contains("cDNA");
+  }
+
+  private Predicate<RunDto> byCreatedBefore(String end) {
+    return dto -> {
+      return start == null && end != null || dto.getCreatedDate().compareTo(end) < 0;
+    };
+  }
+
+  @Override
+  protected List<ColumnDefinition> getColumns() {
+    String analyteHeader = (analyte == null ? "DNA" : analyte);
+    return Arrays.asList(
+        new ColumnDefinition(String.format("%s: %s Recent (%s - %s)", project, analyteHeader, start, end)),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""),
+        new ColumnDefinition(""));
+  }
+
+  @Override
+  protected int getRowCount() {
+    if (analyte == null) {
+      // print DNA and RNA on same sheet
+      return dnaRecentList.size()
+          + 3
+          + rnaRecentList.size()
+          + 3
+          + dnaAllList.size()
+          + 3
+          + rnaAllList.size()
+          + 1;
+    } else if (DNA.equals(analyte)) {
+      // only DNA
+      return dnaRecentList.size()
+          + 3
+          + dnaAllList.size()
+          + 1;
+    } else {
+      // only RNA
+      return rnaRecentList.size()
+          + 3
+          + rnaAllList.size()
+          + 1;
+    }
+    // the "+1" is to make the getRow() call happy -- adds a blank row at the end since
+    // don't know whether the last row will be DNA or RNA
+  }
+
+  @Override
+  protected String[] getRow(int rowNum) {
+    if (rowNum == 0) {
+      return makeHeadingRow();
+    }
+    rowNum -= 1;
+
+    if (analyte == null || analyte.equals(DNA)) {
+      if (rowNum < dnaRecentList.size()) {
+        return makeStatusRow(dnaRecentList.get(rowNum), end);
+      }
+      rowNum -= dnaRecentList.size();
+    }
+
+    if (analyte == null || RNA.equals(analyte)) {
+      if (analyte == null) {
+        // add extra headers for DNA & RNA combined report.
+        // if RNA.equals(analyte) then the "RNA Recent" title and heading are already written
+        if (rowNum == 0) {
+          return makeBlankRow();
+        } else if (rowNum == 1) {
+          return makeTitleRow(String.format("%s: %s (%s - %s)", project, "RNA Recent", start, end));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "%s: %s (%s - %s)" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)